### DISCUSSION TITLE
Want multi select combobox support

### DISCUSSION THREAD

**Discussion by RajaTheKing826** [adobe/react-spectrum#6652] [Q&A]
I know you don't have a requirement so there is no UI to show multi select combobox, but can you please gimme some code to implement multiselect combobox atleast a codesandbox with bare minimal UI. Which is a crucial thing for us.

**Comment by predaytor** [comment]
I am also thinking about it, it should be possible to extend the functionality yourself, it is quite complex stuff, but it's doable. You can look at this:

[ComboBox](https://github.com/adobe/react-spectrum/blob/b46d23b9919eaec8ab1f621b52beced82e88b6ca/packages/react-aria-components/src/ComboBox.tsx)
[useComboBox](https://github.com/adobe/react-spectrum/blob/b46d23b9919eaec8ab1f621b52beced82e88b6ca/packages/%40react-aria/combobox/src/useComboBox.ts)
[useComboBoxState](https://github.com/adobe/react-spectrum/blob/b46d23b9919eaec8ab1f621b52beced82e88b6ca/packages/%40react-stately/combobox/src/useComboBoxState.ts)
[useSingleSelectListState](https://github.com/adobe/react-spectrum/blob/b46d23b9919eaec8ab1f621b52beced82e88b6ca/packages/%40react-stately/list/src/useSingleSelectListState.ts)

How to display the selected items is up to you, I would put a TagGroup next to combobox, similar to Solid.js [Kobalte](https://kobalte.dev/docs/core/components/combobox#multiple-selection) implementation.

**Comment by snowystinger** [comment]
Duplicate of https://github.com/adobe/react-spectrum/issues/2140
